#数据集kitti------------------------------------------------
#话题订阅------------------------------------------------
topic_sub: #
  imu_sub_topic: /kitti/oxts/imu #sensor_msgs/Imu
  cloud_sub_topic: /kitti/velo/pointcloud #sensor_msgs/PointCloud2
  gnss_sub_topic: /kitti/oxts/gps/fix #sensor_msgs/NavSatFix
  ground_truth_sub_topic: / #待定

#局部地图设置-------------------------------------------------
local_map:
  keyframe_distance: 2.0
  keyframe_num: 10

#传感器设置-------------------------------------------------
sensor_setting:
  use_gnss: true

#局部地图设置------------------------------------------------
output_setting:
  save_global_map: true

#滤波器参数------------------------------------------------
voxel_filter_size:
  single_scan: 1.3
  local_map: 0.6
  global_map: 0.5

#空间外参-------------------------------------------------
extrinsic:
  lidar_to_body:
    [
      0.999998,
      -0.000785403,
      0.00202441,
      0.810544,
      0.000755307,
      0.99989,
      0.0148245,
      -0.307054,
      -0.00203583,
      -0.014823,
      0.999888,
      0.802724,
      0.0,
      0.0,
      0.0,
      1.0,
    ]
  imu_to_body:
    [
      1.0,
      0.0,
      0.0,
      0.0,
      0.0,
      1.0,
      0.0,
      0.0,
      0.0,
      0.0,
      1.0,
      0.0,
      0.0,
      0.0,
      0.0,
      1.0,
    ]

#相机内参-------------------------------------------------
camera_intrinsics:
  model_type: PINHOLE
  camera_name: camera
  image_width: 1920
  image_height: 1200
  distortion_parameters:
    k1: -0.109203
    k2: 0.063536
    p1: -0.003427
    p2: -0.000629
  projection_parameters:
    fx: 1086.160899
    fy: 1090.242963
    cx: 940.067502
    cy: 586.740077

#IMU 内参-------------------------------------------------
imu_intrinsics:
  gyr_n: 7.4266825125507141e-03 #高斯噪声
  gyr_w: 8.6572985145653080e-05 #随机游走
  acc_n: 8.1330537434371481e-03
  acc_w: 1.2123362494392119e-04

#GNSS 内参-------------------------------------------------
gnss_intrinsics:
  accuracy: 1.5
# #数据集urbannav------------------------------------------------
# #话题订阅------------------------------------------------
# topic_sub: #
#   imu_sub_topic: /imu/data #sensor_msgs/Imu
#   cloud_sub_topic: /velodyne_points #sensor_msgs/PointCloud2
#   gnss_sub_topic: /navsat/fix #sensor_msgs/NavSatFix
#   ground_truth_sub_topic: / #待定

# #局部地图设置-------------------------------------------------
# local_map:
#   keyframe_distance: 2.0
#   keyframe_num: 20

# #传感器设置-------------------------------------------------
# sensor_setting:
#   use_gnss: true

# #局部地图设置------------------------------------------------
# output_setting:
#   save_global_map: true

# #滤波器参数------------------------------------------------
# voxel_filter_size:
#   single_scan: 1.3
#   local_map: 0.6
#   display: 0.5

# #空间外参-------------------------------------------------
# extrinsic:
#   lidar_to_body:
#     [
#       0.999998,
#       -0.000785403,
#       0.00202441,
#       0.810544,
#       0.000755307,
#       0.99989,
#       0.0148245,
#       -0.307054,
#       -0.00203583,
#       -0.014823,
#       0.999888,
#       0.802724,
#       0.0,
#       0.0,
#       0.0,
#       1.0,
#     ]
#   imu_to_body:
#     [
#       1.0,
#       0.0,
#       0.0,
#       0.0,
#       0.0,
#       1.0,
#       0.0,
#       0.0,
#       0.0,
#       0.0,
#       1.0,
#       0.0,
#       0.0,
#       0.0,
#       0.0,
#       1.0,
#     ]

# #相机内参-------------------------------------------------
# camera_intrinsics:
#   model_type: PINHOLE
#   camera_name: camera
#   image_width: 1920
#   image_height: 1200
#   distortion_parameters:
#     k1: -0.109203
#     k2: 0.063536
#     p1: -0.003427
#     p2: -0.000629
#   projection_parameters:
#     fx: 1086.160899
#     fy: 1090.242963
#     cx: 940.067502
#     cy: 586.740077

# #IMU 内参-------------------------------------------------
# imu_intrinsics:
#   gyr_n: 7.4266825125507141e-03 #高斯噪声
#   gyr_w: 8.6572985145653080e-05 #随机游走
#   acc_n: 8.1330537434371481e-03
#   acc_w: 1.2123362494392119e-04

# #GNSS 内参-------------------------------------------------
# gnss_intrinsics:
#   accuracy: 1.5

